docs/deployment.md

```diff
@@ -105,7 +105,7 @@ FROM node:alpine AS builder
 WORKDIR /app
 COPY . .
 COPY --from=deps /app/node_modules ./node_modules
-RUN yarn build && yarn install --production --ignore-scripts --prefer-offline
+RUN yarn build
```

Comment:
The `COPY` step copies all possible `node_modules` but no all of them are needed after the build.

The `--production` install only installs necessary deps needed at runtime.

Reply:
Actually Next.js needs devDeps, too. Well, at least `@types/react` and `@types/node` that are usually installed in devDeps. So with `--production` flag the build will fail.

See my issue about that [here](https://github.com/vercel/next.js/issues/31736)

And, please, see an example that the links _in the docs_ points to https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile#L14

Reply:
Yes it would fail the build, which is why `yarn build` is run before `yarn install --production`

Reply:
So, you are saying that this line solves the problem of `node_modules` size in the final image? 
Looks like a big hack around Next.js check of `@types/` packages check during the build, don't you think? 😄

Why do we even need `node_modules` after we had built the app fo it to work?

Reply:
I tend to drop the devDependencies after `yarn build` in the builder stage. 

Check this [repo example](https://github.com/belgattitude/nextjs-monorepo-example) to see how it makes a difference (1.9gb -> 820mb)

![image](https://user-images.githubusercontent.com/259798/145660524-403903f6-7b71-4a1a-b9f0-1fd9274d89e0.png) (basic doc [here](https://github.com/belgattitude/nextjs-monorepo-example/blob/main/docs/docker/docker.md))

So I suggest to keep the example like it is --production (or yarn workspace focus).

@vdvukhzhilov 

> Well, at least @types/react and @types/node that are usually installed in devDeps

Strange, I haven't had that problem... Are you sure ?

And/Or wait for the new output trace https://github.com/vercel/next.js/pull/32258 support, cause it looks amazing.

Reply:
@belgattitude 

>  I haven't had that problem

Probably, you haven't had this problem because

> I tend to drop the devDependencies after yarn build in the builder stage.

But the whole point is not to do that in the first place. Because this is redundant for full production builds.